src/services/kartoffel.service.ts

```diff
@@ -0,0 +1,70 @@
+import config from '../config';
+import IKartoffelService from '../interfaces/kartoffelService.interface';
+import axios, { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import getToken from '../auth/spike';
+import * as https from 'https';
+
+class TokenWrap {
+    token?: string;
+
+    async getToken(expired: Boolean): Promise<string> {
+        if (expired || !this.token) {
+            this.token = await getToken();
+            return this.token;
+        }
+
+        return this.token;
+    }
+}
+
+
+class KartoffelService implements IKartoffelService {
+    private _kartoffelAxios: AxiosInstance;
+    private _tokenWrap: TokenWrap;
+
+    constructor(baseUrl: string) {
+        this._kartoffelAxios = axios.create({
+            baseURL: baseUrl,
+            httpsAgent: new https.Agent({ rejectUnauthorized: false })
+        });
+        this._tokenWrap = new TokenWrap();
+        this.setUpAxiosInterceptors();
+    }
+
+    private setUpAxiosInterceptors = async (): Promise<void> => {
+        this._kartoffelAxios.interceptors.request.use(async (reqConfig: AxiosRequestConfig) => {
+            try {
+                if (config.kartoffel.isAuth) reqConfig.headers!.Authorization = await this._tokenWrap.getToken(false);
+                return reqConfig;
+            } catch (err) {
+                console.log(err)
+                return reqConfig;
+            }
+        });
+
+        this._kartoffelAxios.interceptors.response.use(
+            (response: AxiosResponse) => response,
+            async (error: AxiosError) => {
+                if (error.status === '401') {
+                    const { config } = error;
+                    config.headers!.Authorization = await this._tokenWrap.getToken(true);
```

Comment:
this._tokenWrap.getToken(renew=true)

Reply:
why specify the param name ?

Reply:
or not, but it makes it more clear here, "renew" is more accurate